cli/src/main.rs

```diff
@@ -45,58 +76,118 @@ struct Connection {
 
 #[derive(Subcommand)]
 enum Commands {
-    Status,
-    Deploy(DeployArgs),
-}
+    // status
+    Status(status::StatusArgs), //aka topology
+
+    // deployment
+    DeployResource(deploy_resource::DeployResourceArgs),
+
+    // process instance
+    CreateProcessInstance(create_process_instance::CreateProcessInstanceArgs),
+    CancelProcessInstance(cancel_process_instance::CancelProcessInstanceArgs),
 
-#[derive(Args)]
-struct DeployArgs {
-    #[clap(required = true, value_parser, value_name = "FILE")]
-    resources: Vec<PathBuf>,
+    // message
+    PublishMessage(publish_message::PublishMessageArgs),
+
+    // incident
+    ResolveIncident(resolve_incident::ResolveIncidentArgs),
+
+    // variables
+    SetVariables(set_variables::SetVariablesArgs),
+
+    //jobs
+    ActivateJobs(activate_jobs::ActivateJobsArgs),
+    CompleteJob(complete_job::CompleteJobArgs),
+    FailJob(fail_job::FailJobArgs),
+    UpdateRetries(update_retries::UpdateRetriesArgs),
+    ThrowError(throw_error::ThrowErrorArgs),
 }
 
 impl From<Connection> for zeebe_client::Connection {
     fn from(conn: Connection) -> Self {
         match conn.address {
-            Some(addr) => zeebe_client::Connection::Address(addr),
-            None => zeebe_client::Connection::HostPort(conn.host, conn.port),
+            Some(addr) => zeebe_client::Connection::Address {
+                insecure: conn.insecure,
+                addr,
+            },
+            None => zeebe_client::Connection::HostPort {
+                insecure: conn.insecure,
+                host: conn.host,
+                port: conn.port,
+            },
         }
     }
 }
 
+impl TryFrom<Authentication> for zeebe_client::Authentication {
+    type Error = Report;
+
+    fn try_from(auth: Authentication) -> Result<zeebe_client::Authentication> {
+        match (auth.client_id, auth.client_secret) {
+            (None, None) => Ok(zeebe_client::Authentication::Unauthenticated),
+            (Some(client_id), Some(client_secret)) => Ok(zeebe_client::Authentication::Oauth2(
+                zeebe_client::auth::OAuth2Config {
+                    client_id,
+                    client_secret,
+                    auth_server: auth.authorization_server,
+                    audience: "zeebe.camunda.io".to_owned(),
```

Comment:
:x: extract audience from address or allow to set it in command line

Reply:
:+1: I don't think extracting it from the address is appropriate though, it seems to be hard coded in CC and unrelated to the actual address.

Reply:
Honestly, surprised that it does work. According to this: https://github.com/camunda/zeebe/blob/9459549f731bdc85bf492ee45bd8c597c8a12a32/clients/java/src/main/java/io/camunda/zeebe/client/impl/ZeebeClientCloudBuilderImpl.java#L223 audience should include cluster id.

Reply:
There is also this logic in testbench:
```
     if (audience == null) {
        audience = contactPoint.substring(0, contactPoint.lastIndexOf(":"));
      }
```
where contact point is the same as ZEEBE_ADDRESS

Reply:
:+1: You are right, `zeebe.camunda.io` seems to only work for production, other stages require the address based audience.